beautyjoy/BJC-Teacher-Tracker
# Teacher merge feature

## What this PR does:


Implements a feature which allows the admin to merge teachers together. I merged the branches of the other pull requests into this one, so do NOT review this pull request until first merging in [https://github.com/beautyjoy/BJC-Teacher-Tracker/pull/295](https://github.com/beautyjoy/BJC-Teacher-Tracker/pull/295) and [https://github.com/beautyjoy/BJC-Teacher-Tracker/pull/304](https://github.com/beautyjoy/BJC-Teacher-Tracker/pull/304).

This pull request ALSO includes the email template changes made by @perryzjc and @realmichaeltao . I resolved merge conflicts that came up between our two features. To avoid merge conflicts, @cycomachead it would be best if you reviewed this branch for both the features at once. Here is a summary of the features that are included on this branch:

1) The teacher merge feature, see description below.

2) The new email model feature, all relevant PR descriptions are linked here: [https://github.com/beautyjoy/BJC-Teacher-Tracker/pull/308](https://github.com/beautyjoy/BJC-Teacher-Tracker/pull/308), [https://github.com/beautyjoy/BJC-Teacher-Tracker/pull/309](https://github.com/beautyjoy/BJC-Teacher-Tracker/pull/309), [https://github.com/beautyjoy/BJC-Teacher-Tracker/pull/310](https://github.com/beautyjoy/BJC-Teacher-Tracker/pull/310), and [https://github.com/beautyjoy/BJC-Teacher-Tracker/pull/311](https://github.com/beautyjoy/BJC-Teacher-Tracker/pull/311)


This pull request fixes|implements (pick one...) ______.

TEACHER MERGE FEATURE DESCRIPTION BELOW


### Include screenshots, videos, etc.

<img width="1403" alt="Screen Shot 2024-04-15 at 11 04 27 AM" src="https://github.com/beautyjoy/BJC-Teacher-Tracker/assets/65872167/bd80b628-e7a9-4a20-b7df-793b3593d12f">

<img width="1419" alt="Screen Shot 2024-04-15 at 11 04 34 AM" src="https://github.com/beautyjoy/BJC-Teacher-Tracker/assets/65872167/1cbfefa5-f338-453b-93f9-b2ff3363b94d">

<img width="1416" alt="Screen Shot 2024-04-15 at 11 04 42 AM" src="https://github.com/beautyjoy/BJC-Teacher-Tracker/assets/65872167/6eac6e38-ad04-413f-9429-d893e9fcdefa">


#### Who authored this PR?

@ArushC wrote the tests, implemented the frontend design for the merge page and the frontend/backend for the merge button + modal popup on the teacher show page, and setup routes
@razztech worked on backend for the merge function logic, delivered feedback/guidance on frontend design, modified routes for improved clarity.


### How should this PR be tested?

Look at the Cucumber tests to see that everything works.

* Is there a deploy we can view?

Yes: latest deployment at [https://sp24-01-bjc-teachers-d0e972207d82.herokuapp.com/](https://sp24-01-bjc-teachers-d0e972207d82.herokuapp.com/).

* What do the specs/features test?

They show that stuff works...

* Are there edge cases to watch out for?

No

#### Are there any complications to deploying this?

Nope



### Checklist:
